### ISSUE TITLE
Missing `Datadog.h` when installing pods via `npx react-native run-ios`

### ISSUE THREAD

**Issue by sahzhang** [DataDog/dd-sdk-reactnative#232]
**Describe what happened**
Getting `Datadog.h file not found` error not found when running `yarn ios` after a clean installation of `yarn add @datadog/mobile-react-native`

```› Executing @datadog/mobile-react-native Pods/DatadogSDK » Copy generated compatibility header

❌  error: emit-module command failed with exit code 1 (use -v to see invocation)

❌  (/Users/Library/Developer/Xcode/DerivedData/.../Build/Products/Debug-iphonesimulator/DatadogSDK/DatadogSDK-umbrella.h:13:9)

  11 | #endif
  12 | 
> 13 | #import "Datadog.h"
     |         ^ 'Datadog.h' file not found
  14 | #import "ObjcAppLaunchHandler.h"
  15 | #import "ObjcExceptionHandler.h"
  16 | 

› Compiling @datadog/mobile-react-native Pods/DatadogSDKCrashReporting » SwiftExtensions.swift

❌  (/Users/Library/Developer/Xcode/DerivedData/.../Build/Products/Debug-iphonesimulator/DatadogSDK/DatadogSDK-umbrella.h:13:9)

  11 | #endif
  12 | 
> 13 | #import "Datadog.h"
     |         ^ 'Datadog.h' file not found
  14 | #import "ObjcAppLaunchHandler.h"
  15 | #import "ObjcExceptionHandler.h"
  16 | 
  ```

`Pods/DatadogSDKCrashReporting/Sources/Datadog` directory is empty, not sure if this is intended.
**Steps to reproduce the issue:**
- Run `yarn add @datadog/mobile-react-native`
- Run `npx react-native run-ios`

**Describe what you expected:**
Installation to work properly

**Additional context**
    - SDK Version: `^1.0.0-rc9`
    - `react-native` Version: `0.68.2`
    - `react` Version: `17.0.2`

**Comment by louiszawadzki** [comment]
Hi @sahzhang, thank you for reaching out!

Have you run `pod install` after the installation? You might need to run `pod update DatadogSDKBridge` if you made an upgrade to the SDK version.

Could you also try running the project from XCode (open the `ios/YourAppName.xcworkspace` in XCode, then click on the "Play" icon), and let us know if you have the same issue?

And if none of this solved the issue, could you share with us your `Podfile` and `Podfile.lock` to help us troubleshoot?


Thanks a lot :)

**Comment by sahzhang** [comment]
@louiszawadzki, yes I have a run `pod install`. I also tried running `pod update DatadogSDKBridge` and from XCode. All the same issue.

XCode stack trace:
```
<module-includes>:1:9: note: in file included from <module-includes>:1:
#import "DatadogSDK-umbrella.h"
        ^
/Users/Library/Developer/Xcode/DerivedData/.../Build/Products/Debug-iphonesimulator/DatadogSDK/DatadogSDK-umbrella.h:13:9: error: 'Datadog.h' file not found
#import "Datadog.h"
        ^
<unknown>:0: error: could not build Objective-C module 'Datadog'
```

From our `Podfile.lock`:
```
  - DatadogSDK (1.11.1)
  - DatadogSDKBridge (0.5.5):
    - DatadogSDK (~> 1.11.0-rc1)
    - DatadogSDKCrashReporting (~> 1.11.0-rc1)
  - DatadogSDKCrashReporting (1.11.1):
    - DatadogSDK (= 1.11.1)
    - PLCrashReporter (~> 1.10.1)
  - DatadogSDKReactNative (1.0.0-rc9):
    - DatadogSDKBridge (= 0.5.5)
    - React-Core
  ...
  - PLCrashReporter (1.10.2)
  ```

**Comment by sahzhang** [comment]
@louiszawadzki 

We already have a Bridging header file:

```
#import <React/RCTBridge.h>
#import "React/RCTBridgeModule.h"
#import <React/RCTEventEmitter.h>
#import "React/RCTUIManager.h"
#import "React/RCTViewManager.h"
```

**Comment by sahzhang** [comment]
Still getting the same issue. 

I'm on Xcode version 13.4.1. Also tried building the `DatadogSDKCrashReporting` scheme separately and errored out with the same `could not build Objective-C module 'Datadog'`. When just building the `Datadog` scheme, the build passes.

**Comment by louiszawadzki** [comment]
Hi @sahzhang,

On my side I'm able to build the `DatadogSDKCrashReporting` scheme, while my `Pods/DatadogSDKCrashReporting/Sources/Datadog` directory is also empty.

Could you share the output of the `DatadogSDKCrashReporting` scheme build so we can understand at what point it is failing? 
You can get it in the report navigator (you can get there by pressing `⌘9`), then clicking on the "Export" button.

**Comment by sahzhang** [comment]
@louiszawadzki Apologies for the delay, I have the scheme build report. Do you mind if I email this to you?

**Comment by louiszawadzki** [comment]
@sahzhang can you share it in the issue? 
That would help me sharing it with the native iOS SDK team :) 

If you don't feel confident sharing the file here no problems, tell me and I'll contact you through the email address from your github profile.
I'm going to be off in the next 2 weeks, so don't be alarmed if you don't hear back from me :)

**Comment by sahzhang** [comment]
Yes could you please email me. Thanks! :)

**Comment by louiszawadzki** [comment]
Hi @sahzhang!

Sorry for the long delay, I am just back! Actually I think the easiest will be for you to email the file to support@datadoghq.com and mention that we've discussed this in this issue as well as my name (Louis Zawadzki), and it will be forwarded very quickly to me.
Going through support will make it a lot easier for me to share it with our iOS team and to reference the case later :) 

If it turns out to take too much time I'll email you :) 
Thanks a lot!

**Comment by louiszawadzki** [comment]
Hi @sahzhang,

I did not hear from Datadog's support yet, did you manage to reach out to them?
I am thinking that the issue could also come from the `Podfile`, could you share this file with us as well?

Thanks a lot

**Comment by juanjorge** [comment]
hey @louiszawadzki 
here the `Podfile`

```
require File.join(File.dirname(`node --print "require.resolve('expo/package.json')"`), "scripts/autolinking")
require_relative '../node_modules/react-native/scripts/react_native_pods'
require_relative '../node_modules/@react-native-community/cli-platform-ios/native_modules'

platform :ios, '12.0'
install! 'cocoapods', :deterministic_uuids => false

require 'json'
podfile_properties = JSON.parse(File.read('./Podfile.properties.json')) rescue {}

target 'Measured' do
  use_expo_modules!
  post_integrate do |installer|
    begin
      expo_patch_react_imports!(installer)
    rescue => e
      Pod::UI.warn e
    end
  end
  config = use_native_modules!

  # Flags change depending on the env values.
  flags = get_default_flags()

  use_react_native!(
    :path => config[:reactNativePath],
    # to enable hermes on iOS, change `false` to `true` and then install pods
    :hermes_enabled => podfile_properties['expo.jsEngine'] == 'hermes',
    :fabric_enabled => flags[:fabric_enabled],
    # An absolute path to your application root.
    :app_path => "#{Pod::Config.instance.installation_root}/.."
  )

  # Enables Flipper.
  #
  # Note that if you have use_frameworks! enabled, Flipper will not work and
  # you should disable the next line.
  use_flipper!()

  post_install do |installer|
    react_native_post_install(installer)
    __apply_Xcode_12_5_M1_post_install_workaround(installer)
  end

  pod 'RNSVG', :path => '../node_modules/react-native-svg'

  permissions_path = '../node_modules/react-native-permissions/ios'
  
  pod 'Permission-Camera', :path => "#{permissions_path}/Camera"
  pod 'Permission-Microphone', :path => "#{permissions_path}/Microphone"
  pod 'Permission-Notifications', :path => "#{permissions_path}/Notifications"
  pod 'Permission-PhotoLibrary', :path => "#{permissions_path}/PhotoLibrary"
  pod 'RNSentry', :path => '../node_modules/@sentry/react-native'

end
```

I tested out with the sdk: `^1.0.0-rc9` and `1.1.1`

thank you in advanced

**Comment by juanjorge** [comment]
hey @louiszawadzki
I got working this library at the end.

I was upgrading the expo SDK and RN version, I was dealing with some issues, the way I faced out was removed completly from the project and get running the app, fixing all the issues related with the native changes on iOS and Android, and then I did a fresh installation of the DD module and everything works as expected.

thank you for you review.
